docs/user-guide/dhcp.md

```diff
@@ -0,0 +1,346 @@
+
+- Servers requiring consistent IP addresses
+- Network equipment that needs fixed IPs
+- Hosts that need IPs outside the dynamic range
+
+Static leases can be configured in two ways:
+
+1. Via VPC YAML - Configure static leases in the VPC subnet definition
+2. Via DHCPSubnet resource - Directly modify the DHCPSubnet resource
+
+### Method 1: Configuring Static Leases via VPC
+
+```yaml title="vpc-with-static-leases.yaml"
+apiVersion: vpc.githedgehog.com/v1beta1
+kind: VPC
+metadata:
+  name: vpc-1
+  namespace: default
+spec:
+  subnets:
+    default:
+      subnet: 10.10.1.0/24
+      gateway: 10.10.1.1
+      vlan: 1001
+      dhcp:
+        enable: true
+        range:
+          start: 10.10.1.10
+          end: 10.10.1.99
+        static:
+          "aa:bb:cc:dd:ee:01":
+            ip: 10.10.1.5
+          "aa:bb:cc:dd:ee:02":
+            ip: 10.10.1.6
+          "aa:bb:cc:dd:ee:03":
+            ip: 10.10.1.100
+```
+
+Apply the configuration:
+
+```bash
+kubectl apply -f vpc-with-static-leases.yaml
+```
+
+### Method 2: Configuring Static Leases via DHCPSubnet
+
+For advanced use cases, you can directly edit the DHCPSubnet resource. The DHCPSubnet is automatically created when you enable DHCP on a VPC subnet, with the name format `<vpc-name>--<subnet-name>`.
+
+First, find the DHCPSubnet resource:
+
+```bash
+kubectl get dhcpsubnets.dhcp.githedgehog.com
+```
+
+Example output:
+
+```
+NAME                CIDR           START IP      END IP
+vpc-1--default      10.10.1.0/24   10.10.1.10    10.10.1.99
+```
+
+View the current configuration:
+
+```bash
+kubectl get dhcpsubnets.dhcp.githedgehog.com vpc-1--default -o yaml
+```
+
+Edit the DHCPSubnet to add static leases:
+
+```bash
+kubectl edit dhcpsubnets.dhcp.githedgehog.com vpc-1--default
+```
+
+Add the static leases under `spec.static`:
+
+```yaml
+apiVersion: dhcp.githedgehog.com/v1beta1
+kind: DHCPSubnet
+metadata:
+  name: vpc-1--default
+  namespace: default
+spec:
+  subnet: vpc-1/default
+  cidrBlock: 10.10.1.0/24
+  gateway: 10.10.1.1
+  startIP: 10.10.1.10
+  endIP: 10.10.1.99
+  vrf: VrfVvpc-1
+  circuitID: Vlan1001
+  leaseTimeSeconds: 3600
+  static:                           # Add static leases here
+    "aa:bb:cc:dd:ee:01":
+      ip: 10.10.1.5
+    "aa:bb:cc:dd:ee:02":
+      ip: 10.10.1.6
+    "aa:bb:cc:dd:ee:03":
+      ip: 10.10.1.100
+```
+
+Alternatively, use `kubectl patch`:
+
+```bash
+kubectl patch dhcpsubnets.dhcp.githedgehog.com vpc-1--default --type=merge -p '
+{
+  "spec": {
+    "static": {
+      "aa:bb:cc:dd:ee:01": {"ip": "10.10.1.5"},
+      "aa:bb:cc:dd:ee:02": {"ip": "10.10.1.6"}
+    }
+  }
+}'
+```
+
+!!! warning "VPC vs DHCPSubnet Configuration"
+    - Changes made to the VPC YAML will overwrite static leases configured directly in DHCPSubnet
+    - It's recommended to choose one method and stick with it to avoid configuration conflicts
+    - The VPC method is preferred for most use cases as it keeps configuration centralized
```

Comment:
Based on these warnings I am inferring that if a user makes a change in the DHCPSubnet configuration, the use won't be able to see those changes in the VPC object? If that is true, can we add that explicitly here? My concern comes from there being two people making changes and stepping on each others toes. 

I like that you give guidance to use the VPC method.